Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CentralRepository.java

```diff
@@ -449,9 +449,10 @@ CorrelationAttributeInstance getCorrelationAttributeInstance(CorrelationAttribut
      *
      * @param eamArtifact Artifact containing exactly one (1) ArtifactInstance.
      * @param knownStatus The status to change the artifact to
+     * @param isTagDelete If the action is to delete a previously added tag.
      */
-    void setAttributeInstanceKnownStatus(CorrelationAttributeInstance eamArtifact, TskData.FileKnown knownStatus) throws CentralRepoException;
-
+    void setAttributeInstanceKnownStatus(CorrelationAttributeInstance eamArtifact, TskData.FileKnown knownStatus, boolean isTagDelete) throws CentralRepoException;
+    
```

Comment:
@gdicristofaro and @wschaeferB... 

It is not appropriate to add a flag about tag deletion to a method whose stated purpose is to update the known status of a correlation attribute instance (even acknowledging that the method has a completely undocumented side effect of setting comment, not just the known status).

The CentralRepository interface defines the operations that can be performed on the data store. Application / domain logic does not belong in the data store layer of an architecture, it belongs in the application / domain logic layer. In this case, the decision whether or not to change the known status and the comment belongs where it was, in the application logic implemented by the CaseEventListener.

Another way to think about this is like this: the CentralRepository interface exists because we have envisioned having an option to choose from multiple relational and non-relational data store technologies for the CR. Putting the application logic details in CentralRepository interface implementations means that some future implementer of a NoSQL CR will need to **_duplicate_** the complicated logic that has been put into RdbmsCentralRepo (as in Relational Database Management System Central Repo). This would introduce a maintenance problem that would be magnified with every new implementation of CentralRepository.